TRAINING_IN_DETAIL.rst

```diff
@@ -0,0 +1,285 @@
+   file_name = env.hyperparameters['training_data_file']
+
+   # get the folder where the model should be saved
+   model_dir = env.model_dir
+   data = np.load(os.path.join(training_dir, file_name))
+   x_train, y_train = data['features'], keras.utils.to_categorical(data['labels'])
+   model = ResNet50(weights='imagenet')
+   ...
+   model.fit(x_train, y_train)
+
+   #save the model in the end of training
+   model.save(os.path.join(model_dir, 'saved_model'))
+
+**Entrypoint execution** is encapsulted by `entry_point.run(uri, user_entry_point, args, env_vars) <https://github.com/aws/sagemaker-containers/blob/v2.4.4/src/sagemaker_containers/entry_point.py#L22>`__,
+it prepares and executes the user entry point, passing ``env_vars`` as
+environment variables and ``args`` as command arguments. If the entry
+point is:
+
+-  **A Python script:** executes the script as
+   ``ENV_VARS python entry point_name ARGS``
+
+-  **Any other script:** executes the command as
+   ``ENV_VARS /bin/sh -c ./module_name ARGS``
+
+Usage example:
+
+.. code:: python
+
+   import sagemaker_containers
+   from sagemaker_containers.beta.framework import entry_point
+
+   env = sagemaker_containers.training_env()
+   # {'channel-input-dirs': {'training': '/opt/ml/input/training'}, 'model_dir': '/opt/ml/model', ...}
+
+   # reading hyperparameters as a dictionary
+   hyperparameters = env.hyperparameters
+   # {'batch-size': 128, 'model_dir': '/opt/ml/model'}
+
+   # reading hyperparameters as script arguments
+   args = env.to_cmd_args(hyperparameters)
+   # ['--batch-size', '128', '--model_dir', '/opt/ml/model']
+
+   # reading the training environment as env vars
+   env_vars = env.to_env_vars()
+   # {'SAGEMAKER_CHANNELS':'training', 
+   #  'SAGEMAKER_CHANNEL_TRAINING':'/opt/ml/input/training',
+   #  'MODEL_DIR':'/opt/ml/model', ...}
+
+   # executes user entry point named entry point.py as follow:
+   #
+   # SAGEMAKER_CHANNELS=training SAGEMAKER_CHANNEL_TRAINING=/opt/ml/input/training \
+   # SAGEMAKER_MODEL_DIR=/opt/ml/model python user_script.py --batch-size 128 --model_dir /opt/ml/model
+   entry_point.run('entry point.py', args, env_vars)
+
+If the entry point execution fails, ``trainer.train()`` will write the
+error message to ``/opt/ml/output/failure``.
+
+The entry point touches the sucess file under ``/opt/ml/success``
+otherwise.
+
+.. _header-n814:
+
+Training a Framework container
+------------------------------
+
+`TensorFlow <https://github.com/aws/sagemaker-tensorflow-container/tree/script-mode>`__,
+`MXNet <https://github.com/aws/sagemaker-mxnet-container>`__,
+`PyTorch <https://github.com/aws/sagemaker-pytorch-container>`__,
+`Chainer <https://github.com/aws/sagemaker-chainer-container>`__, and
+`Scikit-Learn <https://github.com/aws/sagemaker-scikit-learn-container>`__ are
+**Framework Containers**. One difference between a **Framework
+Container** and a **BYOC** is while the latter includes the entry point
```

Comment:
the difference you're referring to here is actually between an image with SageMaker Containers and one without

Reply:
Not really. The previous example mentions an image using SageMaker Containers that does not need to download the user code from S3.